man/man1/man1.mk

```diff
@@ -34,4 +34,16 @@ if ENABLE_P11SAK
 man1_MANS += man/man1/p11sak.1
 endif
 
+if ENABLE_PKCSEP11_MIGRATE
+man1_MANS += man/man1/pkcstok_admin.1
+
+man/man1/pkcstok_admin.1: man/man1/pkcstok_admin.1.in
+	$(AM_V_GEN)@SED@ -e s!\@sysconfdir\@!"$(sysconfdir)"!g		\
+			 -e s!\@pkcs_group\@!"$(pkcs_group)"!g		\
+			 -e s!\@localstatedir\@!"$(localstatedir)"!g 	\
+			 < $< > $@-t &&					\
+	$(am__mv) $@-t $@
+endif
+
+EXTRA_DIST += man/man1/pkcstok_admin.1.in
```

Comment:
Shouldn't this be within the `if ENABLE_PKCSEP11_MIGRATE` block?

Reply:
No actually not. Note that `man/man1/pkcstok_admin.1.in` is the source file, not the generated man page. The distribution (built via `make dist`) must contain all source files (independent of the current configuration), but should not contain any generated files (Note that we incorrectly distributed the generated make files until some time ago. This was fixed with https://github.com/opencryptoki/opencryptoki/commit/8b31762ad9376a2440e6b04ab65bfa1b28bd4850 ).

For the other man pages this works automatically, because they are mentioned in a `AC_CONFIG_FILES` although this is within a `AM_COND_IF`. It still automatically includes the source files of these man pages into the distribution, even if the condition is not met (i.e. it includes the source files independent of the feature being enabled or not).

However, `man/man1/pkcstok_admin.1.in` is different, because it is not processed via `AC_CONFIG_FILES`, but via a manual `sed` step within `man1.mk`. This is because `AC_CONFIG_FILES` does not replace all placeholders the way it is needed (i.e. `\@sysconfdir\@` would result in something like `${prefix}/...`, but not the full path.  So, because `man/man1/pkcstok_admin.1.in` is not mentioned in a `AC_CONFIG_FILES`, it must be explicitly added to `EXTRA_DIST`. And since source files mut be distributed independently from the current configuration, this has to be outside of the if block. Will fix this.

But your comment still lead to me identifying a bug. The if block of the sed rule should of course be for `ENABLE_PKCSTOK_ADMIN` and not for `ENABLE_PKCSEP11_MIGRATE` (copy/paste error).